platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -71,81 +79,21 @@
-                () ->
-                    Collections.unmodifiableList(
-                        Arrays.asList(
-                            StringUtils.split(System.getProperty(HTTPS_CIPHER_SUITES, ""), ","))));
-  }
-
-  public static final String DEFAULT_SCHEMA_XML = "schema.xml";
-
-  public static final String DEFAULT_SOLRCONFIG_XML = "solrconfig.xml";
-
   private static final Logger LOGGER = LoggerFactory.getLogger(HttpSolrClientFactory.class);
 
-  @Override
-  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
-    String solrUrl =
-        StringUtils.defaultIfBlank(
-            AccessController.doPrivileged(
-                (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
-            getDefaultHttpsAddress());
-    final String coreUrl = solrUrl + "/" + core;
-    final String solrDataDir =
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty(SOLR_DATA_DIR));
+  private final SolrPasswordUpdate solrPasswordUpdate;
+  private final EncryptionService encryptionService;
 
-    if (solrDataDir != null) {
-      ConfigurationStore.getInstance().setDataDirectoryPath(solrDataDir);
-    }
-    LOGGER.debug("Solr({}): Creating an HTTP Solr client using url [{}]", core, coreUrl);
-    return new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+  public HttpSolrClientFactory(
+      EncryptionService encryptionService, SolrPasswordUpdate solrPasswordUpdate) {
+    this.encryptionService = encryptionService;
+    this.solrPasswordUpdate = solrPasswordUpdate;
   }
 
   @VisibleForTesting
-  SolrClient createSolrHttpClient(String url, String coreName, String coreUrl)
-      throws IOException, SolrServerException {
-    final HttpClientBuilder builder =
-        HttpClients.custom()
-            .setDefaultCookieStore(new BasicCookieStore())
-            .setMaxConnTotal(128)
-            .setMaxConnPerRoute(32);
-
-    if (StringUtils.startsWithIgnoreCase(url, "https")) {
-      builder.setSSLSocketFactory(
-          new SSLConnectionSocketFactory(
-              getSslContext(),
-              getProtocols(),
-              getCipherSuites(),
-              SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER));
-    }
-    createSolrCore(url, coreName, null, builder.build());
-    try (final Closer closer = new Closer()) {
-      final HttpSolrClient noRetryClient =
-          closer.with(new HttpSolrClient.Builder(coreUrl).withHttpClient(builder.build()).build());
-      final HttpSolrClient retryClient =
-          closer.with(
-              new HttpSolrClient.Builder(coreUrl)
-                  .withHttpClient(
-                      builder.setRetryHandler(new SolrHttpRequestRetryHandler(coreName)).build())
-                  .build());
-
-      return closer.returning(new PingAwareSolrClientProxy(retryClient, noRetryClient));
-    }
+  HttpSolrClientFactory() {
+    solrPasswordUpdate = null;
```

Comment:
✏️ I would suggest calling the previous constructor with `(null, null)`

Reply:
changed constructor implementation